app/frontend/src/features/search/useSearchFilters.ts

```diff
@@ -0,0 +1,88 @@
+import { Location } from "history";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { useHistory, useLocation } from "react-router-dom";
+import { UserSearchFilters as ServiceUserSearchFilters } from "service/search";
+
+export interface SearchFilters extends ServiceUserSearchFilters {
+  location?: string;
+}
+
+export default function useSearchFilters(route: string) {
+  const location = useLocation();
+  const [active, setActive] = useState<SearchFilters>(
+    locationToFilters(location)
+  );
```

Comment:
```suggestion
  const [active, setActive] = useState<SearchFilters>(
    () => locationToFilters(location)
  );
```
So `locationToFilters` is lazily evaluated and only run once for initialising the state only